version: "3.8"
services:
  client:
    build:
      context: .
      dockerfile: client/Dockerfile.dev
    volumes:
      - ./client:/app/client:delegated
      - ./shared:/app/shared:delegated
      - /app/client/node_modules
      - /app/shared/node_modules
    ports:
      - "127.0.0.1:8080:8080"
    healthcheck:
      test: curl --fail http://localhost:8080 || exit 1
      interval: 5s
      timeout: 5s
      retries: 10
  server:
    build:
      context: .
      dockerfile: server/Dockerfile.dev
    volumes:
      - ./server:/app/server:delegated
      - ./shared:/app/shared:delegated
      - /app/server/node_modules
      - /app/shared/node_modules
    ports:
      - "127.0.0.1:3003:3003"
    depends_on:
      db:
        condition: service_healthy
    healthcheck:
      test: curl --fail http://localhost:3003/api/v1/health || exit 1
      interval: 5s
      timeout: 5s
      retries: 10
  db:
    image: postgis/postgis:13-3.3-alpine
    restart: always
    environment:
      POSTGRES_USER: db_dev_user
      POSTGRES_PASSWORD: DevPassword
      POSTGRES_DB: db_dev
    volumes:
      - db-volume:/var/lib/postgresql/data
    ports:
      - "127.0.0.1:5432:5432"
    healthcheck:
      test: psql -U db_dev_user -d db_dev -c "SELECT 1"
      interval: 5s
      timeout: 5s
      retries: 10

volumes:
  db-volume: